srcs_mandatory/builtin_echo.c

```diff
@@ -0,0 +1,106 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   builtin_echo.c                                     :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: choiejae <choiejae@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/03 15:25:58 by ejachoi           #+#    #+#             */
+/*   Updated: 2023/02/06 21:41:55 by choiejae         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+extern int	g_exit_code;
+
+int	count_env_key_length(char *str)
+{
+	int count;
+
+	count = 0;
+	while (*str)
+	{
+		if (ft_isalnum(*str++))
+			count++;
+	}
+	return (count);
+}
+
+int	check_env_echo(char *str, t_config *config)
+{
+	t_list	*cur;
+	t_env	*env;
+	char	**env_key;
+	char	key[128];
+	int		idx_count;
+	
+	env_key = ft_split(str, '$');
+	if (env_key[0][0] == '?')
+	{
+		ft_printf("%d", g_exit_code);
+		ft_free_split(&env_key);
+		return (1);
+	}
+	idx_count = count_env_key_length(env_key[0]);
+	key[0] = '\0';
+	ft_strlcpy(key, env_key[0], idx_count + 1);
+	cur = get_env_list(config->head, key);
+	if (cur)
+	{
+		env = (t_env *)cur->content;
+		ft_printf("%s%s%s", GREEN, env->value, RESET);
+	}
+	ft_free_split(&env_key);
+	return idx_count;
+}
+
+int	check_opt_flag(char *str, int *idx)
+{
+	if (ft_strnstr(str, "-n", 3))
+	{
+		++*idx;
```

Comment:
`++*idx`는 `++`와 `*` 중 `++` 연산자가 더 우선이라 `*(++idx)`와 같은 동작을 하게됩니다.
인덱스 값 증가 목적이시라면 아래처럼 수정 필요해요.
```suggestion
		++(*idx);
```